### PR TITLE
[6.0.0] Autolykos 2 validation for custom message

### PR DESCRIPTION
**Pull request by kushti**
In this PR, Global.powHit() method implemented, to check PoW for custom variant of Autolykos2 hash function

### CODE REVIEW

**Comment by aslesarenko** [comment]
@kushti is this ready for review? Asking because of the failing tests.

**Comment by kushti** [comment]
> @kushti is this ready for review? Asking because of the failing tests.

Makes sense to start with parent branch PR https://github.com/ScorexFoundation/sigmastate-interpreter/pull/968

**Review by aslesarenko — APPROVED** [review]
LGTM, and also see comments.

File: sc/shared/src/main/scala/sigma/compiler/ir/wrappers/sigma/impl/SigmaDslImpl.scala [hunk 1]
```diff
@@ -2169,6 +2177,14 @@ object SigmaDslBuilder extends EntityObject("SigmaDslBuilder") {
         true, true, element[Coll[Byte]]))
     }
 
+    def powHit(k: Ref[Int], msg: Ref[Coll[Byte]], nonce: Ref[Coll[Byte]], h: Ref[Coll[Byte]], N: Ref[Int]): Ref[BigInt] = {
+      println(SigmaDslBuilderClass.getDeclaredMethods().mkString(", "))
```
[the remark this answers is not in the document]
Comment (kushti) [reply sc/shared/src/main/scala/sigma/compiler/ir/wrappers/sigma/impl/SigmaDslImpl.scala:2181]: removed

File: data/shared/src/main/scala/sigma/ast/CostKind.scala [hunk 2]
```diff
@@ -52,5 +54,39 @@ abstract class TypeBasedCost extends CostKind {
   * See [[EQ]], [[NEQ]]. */
 case object DynamicCost extends CostKind
 
+/** Cost of:
+  * 1) converting numeric value to the numeric value of the given type, i.e. Byte -> Int
+  * NOTE: the cost of BigInt casting is the same in JITC (comparing to AOTC) to simplify
```
[the remark this answers is not in the document]
Comment (kushti) [reply data/shared/src/main/scala/sigma/ast/CostKind.scala:59]: fixed

**Pull request merged**